redisdb/datadog_checks/redisdb/redisdb.py

```diff
@@ -292,22 +293,55 @@ def _check_key_lengths(self, conn, instance, tags):
                         continue
 
                     if key_type == 'list':
-                        lengths[text_key] += db_conn.llen(key)
+                        keylen = db_conn.llen(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
                     elif key_type == 'set':
-                        lengths[text_key] += db_conn.scard(key)
+                        keylen = db_conn.scard(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
                     elif key_type == 'zset':
-                        lengths[text_key] += db_conn.zcard(key)
+                        keylen = db_conn.zcard(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
                     elif key_type == 'hash':
-                        lengths[text_key] += db_conn.hlen(key)
+                        keylen = db_conn.hlen(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
+                    elif key_type == 'string':
+                        # Send 1 if the key exists as a string
+                        lengths[text_key]["length"] += 1
+                        lengths_overall[text_key] += 1
                     else:
                         # If the type is unknown, it might be because the key doesn't exist,
                         # which can be because the list is empty. So always send 0 in that case.
-                        lengths[text_key] += 0
-
-        # send the metrics
-        for key, total in iteritems(lengths):
-            self.gauge('redis.key.length', total, tags=tags + ['key:' + key])
+                        lengths[text_key]["length"] += 0
+                        lengths_overall[text_key] += 0
+
+                    # Tagging with key_type since the same key
+                    #   can exist with  different key_type per db
+                    lengths[text_key]["key_type"] = key_type
+
+            # Send the metrics for each db in the redis instance.
+            for key, total in iteritems(lengths):
+                # Only send non-zeros if tagged per db.
+                if total["length"] > 0:
```

Comment:
Is that to prevent sending the metric for unknown key types? Would there be valid cases where we should submit 0 here?

Reply:
This is to prevent affecting the values we get for existing keys and there's really no valid case to submit 0s on a per db basis. We'll still submit a 0 if and only if the key did not exist in all db0-db15.

eg. let's say keyA exists with length of 5 in redis_db:db0. I wouldnt want to send keyA=0 for db1-db15 because being gauges,  the 15 zeros would affect the sibgle value of 5 using average by defaultbin the UI.

say keyB doesn't exits at all, this would prevent sending 0s tagged with each db0-db15. later at line 341, we still send a length of 0 w/o the keytype or db tag. simply because that's how it works prior to this PR